src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -68,7 +69,36 @@ public BusinessesService(List<Listing> allBusinesses) {
     return allBusinesses;
   }
   
-  public List<Listing> getBusinessesFromPlacesApi(MapLocation mapLocation) {
+  public List<Listing> 
+        getBusinessesFromTextSearch(MapLocation mapLocation, String product) {
+    LatLng latLng = 
+          new LatLng(mapLocation.lat, mapLocation.lng);
+    final GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    TextSearchRequest request = PlacesApi.textSearchQuery(context, product);
+    try {
```

Comment:
nit: Please add a space above this line to improve readability.

Reply:
Done